### ISSUE TITLE
Messages don't show up in the chat box, seemingly for no reason

### ISSUE THREAD

**Issue by DamianX** [d3athrow/vgstation13#13504]
#### Description of issue
Some chat messages are not received, but they should be.
I noticed this happening since the MC change was merged.

#### Difference between expected and actual behavior
I expected to read all messages, but some messages don't show up.

#### Steps to reproduce 
>say something
>it doesn't show up in your chatbox
>other person replies anyway, implying they read the message

OR, for example, I witnessed this:
>AI welcomes everyone that arrives onboard
>they answer "Thank you"
>someone says thank you but in my chat there's no "welcome" message
>ahelped and an admin confirmed that indeed, the AI said welcome, but it didn't show up for me

That said, I don't know how you would even try to reproduce this.

#### Specific information for locating

I haven't saved or screenshotted any of this, I believe, but I could try.

#### Length of time in which bug has been known to occur

Since the "Rebasing to TG" merge.

#### Server revision

61aee611bf36644523de3b14b7f09930faa24a3e 

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

**Comment by PJB3005** [comment]
This has always been happening.

I've experienced this BYOND bug ever since beginning to play SS13, sometimes chat messages just don't get through. You'd think goonchat fixed this but nope!

**Comment by DamianX** [comment]
Is there a dupe issue then, or is it just something you've all come to accept?
Also, it may just be my impression, but something may have changed since that PR was merged. Like, before you had the speaking balloon on top of the person speaking, but the message in chat wouldn't appear, and none of the players would see it. - whereas now you don't even get the balloon anymore if you don't get the chat message, but some players do read the message and some don't.

I may be wrong, but this is how I perceive it.

**Comment by Exxion** [comment]
Yes there is another issue on this matter, but I'm pretty sure it got closed because we can't do shit

**Comment by DamianX** [comment]
How does the chat work? Is it documented somewhere? I'd like to know because reliable messaging is a solved problem, and while I'm sure BYOND is not helping, I find it hard to believe this can't be worked around.

**Comment by Exxion** [comment]
Unless we implement our own entire messaging protocol in the game, it can't

**Comment by kol1th** [comment]
I've only seen it happen on vg 🤷‍♂️

**Comment by PJB3005** [comment]
It's 100% a networking issue, which we can't do Shit about.

I mean theoretically you could set up a HTTP server and let goonchat pull from that but that's just dumb.

**Comment by DamianX** [comment]
Why is it dumb? It comes with the benefit of being able to resume the chat when you disconnect, too.

**Comment by PJB3005** [comment]
Much worse performance, latency, and server side cost.

**Comment by eplgr** [comment]
@PJB3005 Is it?
I've recently found out that if you type certain characters, the message doesn't show up for anyone
Examples: 
# `
#  ñ

Which are both right to the "L" key, in my case, because my language uses them. Typos happen. It surely happens to many people. 
Is it possible to "upgrade"/change the chat?

**Comment by DamianX** [comment]
Isn't that related to #13537 ?

**Comment by Exxion** [comment]
Yes, and is also already reported somewhere

**Comment by Intigracy** [comment]
This wasn't happening with anywhere near the frequency it is.

Something changed on our end.

**Comment by Exxion** [comment]
This used to happen this frequently.
Then it stopped.
Then it started again.
No idea exactly why, but I think it probably has more to do with lag than anything.

**Comment by jwhitak** [comment]
Still a bug. Runechat still appears in these cases.

**Comment by west3436** [comment]
Ancient evil chat bug that all codebases suffer from that many have tried, but failed, to defeat. There is no reason to track this here as it's a Byond bug.
